File: frontend/projects/api/src/lib/events/create-event-request.ts
export interface CreateEventRequest {
  readonly title: string;
  readonly startsAtUtc: string;
  readonly endsAtUtc: string | null;
  readonly location: string;
  readonly description: string | null;
  readonly allowPlusOne: boolean;
  readonly showGuestList: boolean;
  readonly inviteeEmails: ReadonlyArray<string> | null;
}

File: backend/src/PullUp.Api/Requests/CreateEventRequest.cs
namespace PullUp.Api.Requests;

public sealed record CreateEventRequest(
    string Title,
    DateTimeOffset StartsAtUtc,
    DateTimeOffset? EndsAtUtc,
    string Location,
    string? Description,
    bool AllowPlusOne,
    bool ShowGuestList,
    IReadOnlyList<string>? InviteeEmails);
